076-decouple-ca-management-from-end-entity-certificate-management.md

```diff
@@ -0,0 +1,389 @@
+3. The CaReconciler generates the Cluster Operator certificate and stores it in the <CLUSTER_NAME>-cluster-operator-certs Kubernetes Secret.
+   It does this even though the state is UNTRUSTED, because the Kubernetes Secret was missing.
+4. __The CaReconciler places the Cluster CA certificate in the trusted-certs Kubernetes Secret.__
+5. The ZooKeeper and Kafka reconcilers:
+   1. Generate their certificates and store them in the <CLUSTER_NAME>-kafka-brokers (for Kafka brokers) <CLUSTER_NAME>-zookeeper-nodes (for ZooKeeper nodes) Kubernetes Secrets. 
+   2. Create the StrimziPodSets for ZooKeeper/Kafka and add the cluster-ca-cert-generation and server-cert-hash annotations.
+6. The EntityOperator, CruiseControl and KafkaExporter reconcilers generate their certificates and create their Kubernetes Deployments.
+7. A new reconciliation starts.
+8. __The CaReconciler sees that the cluster-ca-cert-generation on the ZooKeeper and Kafka Kubernetes Pods is current so updates the state of the Cluster CA in the trust-state Kubernetes Secret to TRUSTED_IN_USE.__
+
+#### Fig 5: New cluster create flow
+
+![New cluster create flow](./images/076-cert-new-proposed.png)
+
+> Fig 5: The numbers in the diagram match the numbers in the full description of the steps.
+> The diagram omits the use of annotations and the EntityOperator, CruiseControl and KafkaExporter reconcilers.
+
+When the Cluster CA private key is updated by the user, either by using the force-replace annotation, or the ca-cert-generation annotation:
+
+1. If Strimzi is managing the Cluster CA, the CaReconciler renews the Cluster CA certificate and key. The old certificate and key can be removed from the Kubernetes Secret.
+2. __The CaReconciler takes the new Cluster CA certificate and places it in the trust-state Kubernetes Secret with the state UNTRUSTED. It also updates the old CA certificate to have the state PHASE_OUT.__
+3. __The CaReconciler places the new Cluster CA certificate in the trusted-certs Kubernetes Secret.__
+4. The CaReconciler rolls the Kafka and ZooKeeper Kubernetes Pods to trust the new CA certificate, __then it updates the trust-state Kubernetes Secret to put the new CA certificate into TRUSTED_UNUSED.__
+5. The ZooKeeper and Kafka reconcilers:
+   1. Generate new certificates signed by the new CA key.
+   2. Update the cluster-ca-cert-generation and server-cert-hash annotations on the StrimziPodSets.
+   3. During their rolling update step, roll the Kubernetes Pods for ZooKeeper and Kafka to use the new certificates (this is triggered by diffing the server-cert-hash annotation on the StrimziPodSet versus the individual Pods).
+6. The EntityOperator, CruiseControl and KafkaExporter reconcilers generate new certificates signed by the new CA key and roll the Kubernetes Pods to use the new certificates.
+7. A new reconciliation starts.
+8. __The CaReconciler sees there is a CA certificate in the state TRUSTED_UNUSED in the trust-state Kubernetes Secret.
```

Comment:
How does it see this?

Reply:
During the reconciliation the CaReconciler will use the K8s client to get the Secret and examine it